### ISSUE TITLE
[BUG] Documentation for creating a GaNDLF model

### ISSUE THREAD

**Issue by hasan7n** [mlcommons/medperf#436]
Following the [documentation of creating a GaNDLF model MLCube](https://docs.medperf.org/mlcubes/gandlf_mlcube/) works, but the resulting MLCube doesn't seem to work.

The root of the problem might be that the trained model itself, before being packaged as an MLCube, is faulty.

**Comment by AlexanderGetka-cbica** [comment]
What additional information do we have? Do we get any output from the MLCube when invoked as MedPerf would?

**Comment by hasan7n** [comment]
Thanks for mentioning that! I did the experiment yesterday and thought I lost the logs. Just found them: This is the output from the gandlf command:

```
The ``converters`` are currently experimental. It may not support operations including (but not limited to) Functions in ``torch.nn.functional`` that involved data dimension
WARNING: Initializing 'norm_type' as 'batch'
WARNING: 'batch_norm' is no longer supported, please use 'norm_type' in 'model' instead
WARNING: Initializing 'verbose' as False
WARNING: Initializing 'medcam_enabled' as False
WARNING: Initializing 'save_training' as False
WARNING: Initializing 'pin_memory_dataloader' as False
WARNING: Initializing 'enable_padding' as False
WARNING: Initializing 'scaling_factor' as 1
WARNING: Initializing 'clip_grad' as None
WARNING: Initializing 'track_memory_usage' as False
WARNING: Initializing 'memory_save_mode' as False
WARNING: Initializing 'print_rgb_label_warning' as True
WARNING: Initializing 'grid_aggregator_overlap' as crop
WARNING: Initializing 'determinism' as False
WARNING: Initializing 'clip_mode' as None
WARNING: Setting default step_size to: 0.0002
Current model type :  torch
Number of dims     :  3
Number of channels :  3
Number of classes  :  0
Since Device is CPU, Mixed Precision Training is set to False
ERROR: The 'best_file' was not found
```

**Comment by AlexanderGetka-cbica** [comment]
GaNDLF is  looking for the following:
os.path.join(modelDir, parameters["model"]["architecture"], "_best.pth.tar")
modelDir is always passed in the deployed MLCube as /embedded_model.
this reduces in the tutorial case to 
"/embedded_model/vgg16_best.pth.tar"

In my tests, this path does exist in the container, and I can verify it is a tar archive and not, say, a symlink.

I can investigate more later tonight, but as an initial sanity check, can you shell into the produced container and see if it's present or if there is perhaps something unexpected happening?

`docker run -it --rm --entrypoint="/bin/bash" johnsmith/gandlf_model:0.0.1`

**Comment by hasan7n** [comment]
You are right, I inspected the container contents and it was not there.
A question: shouldn't this `best file` be also already present in the `trained_model_output` directory just after [this step1](https://docs.medperf.org/mlcubes/gandlf_mlcube/#1-train-a-gandlf-model)?

**Comment by hasan7n** [comment]
there is a `parameters.pkl` which I believe it should be our file?

**Comment by AlexanderGetka-cbica** [comment]
Yup, it should be there as well immediately after the training process completes.

**Comment by AlexanderGetka-cbica** [comment]
Oops, didn't see the last response when I posted that. **Both** files should be present in the trained model directory.

Not sure exactly what could be happening where gandlf_run isn't producing the best_file unless training fails to complete.
pinging @sarthakpati for any insight. I'll see what I can find tomorrow about this as well.
